File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/chain_responsibility/MajorManager.java
package jv.com.tencent.cain.design_pattern.chain_responsibility;

/**
 * @author : jiangyu
 * @date : 2020/8/8
 * @desc : xxx
 */
public class MajorManager extends AbstractManager {

    public MajorManager(String name) {
        super(name);
    }

    @Override
    void handleRequest(Request request) {
        if (request.getRequestType() == Request.REQUEST_TYPE_VACATION && request.getRequestCount() <= 5) {
            System.out.println(request.getRequestContent() + ", " + request.getRequestCount() + ", " + name + "已批准");
        } else if (superior != null) {
            superior.handleRequest(request);
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/chain_responsibility/MajorManager.kt
package kt.com.tencent.cain.desigin_pattern.chain_responsibility

/**
 * @author : jiangyu
 * @date   : 2020/8/8
 * @desc   : xxx
 */
class MajorManager(name: String) : Manager(name) {

    override fun handleRequest(request: Request) {
        if (request.requestType == Request.REQUEST_TYPE_VACATION && request.requestCount <= 5) {
            println("${request.requestContent}, ${request.requestCount}, ${name}已批准")
            return
        }
        superior?.handleRequest(request)
    }
}